src/Tools/dotnet-monitor/Exceptions/ExceptionsStore.cs

```diff
@@ -26,10 +28,17 @@ internal sealed class ExceptionsStore : IExceptionsStore, IAsyncDisposable
         private readonly CancellationTokenSource _disposalSource = new();
         private readonly KeyedCollection<ulong, ExceptionInstance> _instances = new ExceptionInstanceCollection();
         private readonly Task _processingTask;
+        private ExceptionsConfigurationSettings _configuration;
 
         private long _disposalState;
 
-        public ExceptionsStore(IEnumerable<IExceptionsStoreCallbackFactory> factories)
+        public ExceptionsStore(IEnumerable<IExceptionsStoreCallbackFactory> factories, IOptions<ExceptionsOptions> options)
+            : this(factories, options.Value.CollectionFilters ?? new())
```

Comment:
Let's push the unrolling of the options into factory, replace this ctor with one that just takes the factories (and update the tests that are passing in an empty configuration to call this ctor), and have the factory call the second ctor.

Reply:
I'm not sure I'm understanding what you have in mind here - by pushing the responsibility of unrolling the options to the `IExceptionsStoreCallbackFactory`, won't we repeatedly do this for each factory (instead of once for the `IEnumerable` of factories)? Additionally, I'm unclear on "have the factory call the second ctor" - if we're expecting an `IEnumerable` of factories in the second ctor, I wouldn't expect us to be calling that from the factory itself. Let me know if I'm misinterpreting your comment and I'll go ahead and refactor it

Reply:
Sorry, I forgot that `ExceptionsStore` is in the service container thus it would need the `IOptions<>` instance on it directly. Please disregard.